dev/com.ibm.ws.grpc_fat/fat/src/com/ibm/ws/fat/grpc/GrpcMetricsTest.java

```diff
@@ -0,0 +1,188 @@
+
+package com.ibm.ws.fat.grpc;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestName;
+import org.junit.runner.RunWith;
+
+import com.gargoylesoftware.htmlunit.WebClient;
+import com.gargoylesoftware.htmlunit.html.HtmlForm;
+import com.gargoylesoftware.htmlunit.html.HtmlPage;
+import com.gargoylesoftware.htmlunit.html.HtmlSubmitInput;
+import com.gargoylesoftware.htmlunit.html.HtmlTextInput;
+import com.ibm.websphere.simplicity.ShrinkHelper;
+import com.ibm.websphere.simplicity.log.Log;
+
+import componenttest.annotation.Server;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.FATServletClient;
+
+@RunWith(FATRunner.class)
+public class GrpcMetricsTest extends FATServletClient {
+
+    protected static final Class<?> c = GrpcMetricsTest.class;
+
+    @Rule
+    public TestName name = new TestName();
+
+    @Server("HelloWorldServer")
+    public static LibertyServer grpcMetricsServer; //GrpcMetricsServer
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        // add all classes from com.ibm.ws.grpc.fat.helloworld.service and io.grpc.examples.helloworld
+        // to a new app HelloWorldService.war
+        ShrinkHelper.defaultDropinApp(grpcMetricsServer, "HelloWorldService.war",
+                                      "com.ibm.ws.grpc.fat.helloworld.service",
+                                      "io.grpc.examples.helloworld");
+
+        // add all classes from com.ibm.ws.grpc.fat.helloworld.client, io.grpc.examples.helloworld,
+        // and com.ibm.ws.fat.grpc.tls to a new app HelloWorldClient.war.
+        ShrinkHelper.defaultDropinApp(grpcMetricsServer, "HelloWorldClient.war",
+                                      "com.ibm.ws.grpc.fat.helloworld.client",
+                                      "io.grpc.examples.helloworld",
+                                      "com.ibm.ws.fat.grpc.tls");
+
+        grpcMetricsServer.startServer(GrpcMetricsTest.class.getSimpleName() + ".log");
+        assertNotNull("CWWKO0219I.*ssl not recieved", grpcMetricsServer.waitForStringInLog("CWWKO0219I.*ssl"));
+    }
+
+    @AfterClass
+    public static void tearDown() throws Exception {
+        grpcMetricsServer.stopServer();
+    }
+
+    /**
+     * Tests a basic gRPC helloworld app. HelloWorldClient.war contains a servlet that can be
+     * used to connect to a gRPC service that will deployed via HelloWorldService.war.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testHelloWorld() throws Exception {
```

Comment:
can you give this test a more specific name?  having duplicate test names could cause confusion in the final junit report